src/targets/gpu/include/migraphx/gpu/gemm_impl.hpp

```diff
@@ -24,28 +24,75 @@
 #ifndef MIGRAPHX_GUARD_RTGLIB_GEMM_IMPL_HPP
 #define MIGRAPHX_GUARD_RTGLIB_GEMM_IMPL_HPP
 
+#include <iterator>
 #include <migraphx/shape.hpp>
 #include <migraphx/argument.hpp>
+#include <migraphx/generate.hpp>
 #include <migraphx/gpu/context.hpp>
+#include <migraphx/reduce_dims.hpp>
+#include <migraphx/gpu/hip.hpp>
+#include <migraphx/time.hpp>
+
+using milliseconds = std::chrono::duration<double, std::milli>;
+using microseconds = std::chrono::duration<double, std::micro>;
 
 namespace migraphx {
 inline namespace MIGRAPHX_INLINE_NS {
 namespace gpu {
 
-void gemm(context& ctx,
-          const shape& output_shape,
-          const std::vector<argument>& args,
-          float alpha,
-          float beta,
-          bool int8_x4_format,
-          bool compute_fp32);
-void gemm(context& ctx,
-          const shape& output_shape,
-          const std::vector<argument>& args,
-          int32_t alpha,
-          int32_t beta,
-          bool int8_x4_format,
-          bool compute_fp32);
+#if ROCBLAS_VERSION_MAJOR >= 2 && ROCBLAS_VERSION_MINOR >= 38
+using flag_type = rocblas_gemm_flags;
+#else
+using flag_type = int;
+#endif
```

Comment:
This can be removed from the header and moved to cpp file.

Reply:
done